cmake/HPX_SetupTarget.cmake

```diff
@@ -199,13 +199,7 @@ function(hpx_setup_target target)
 
   if(NOT target_NOLIBS)
     if(HPX_WITH_DYNAMIC_HPX_MAIN AND ("${_type}" STREQUAL "EXECUTABLE"))
-      if("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux")
-        set(hpx_libs hpx_wrap)
-        set_target_properties(${target} PROPERTIES LINK_FLAGS "-Wl,-wrap=main")
-      elseif(APPLE)
-        set(hpx_libs hpx_wrap)
-        set_target_properties(${target} PROPERTIES LINK_FLAGS "-Wl,-e,_initialize_main")
-      endif()
```

Comment:
There is no explicit need, but the idea was to have the flags propagated through the interface of `hpx_wrap`, i.e. by linking to `hpx_wrap` a target would get those flags set correctly. If that happens there's no need to set the flags explicitly like this (although it wouldn't hurt either). With newer CMakes this will definitely work, but it seems like we might have to keep this around a bit longer to make it work correctly with older CMakes.

Reply:
It would be better to simply bump our minimal cmake requirement to whatever is needed to make things work cleanly and without 'fixes'.
The modularization work is a major breaking change to the old code base, so bumping requirements should be allowed.

Reply:
I'm also in favor of bumping if we need to but in this case it turned out to be a false alarm. In addition, the CMake changes are intended to be *non*-breaking changes. With this PR we get back to non-breaking (modulo corner cases that we haven't caught yet).